test/app-luatest/preload_test.lua

```diff
@@ -0,0 +1,321 @@
+local fun = require('fun')
+local fio = require('fio')
+local log = require('log')
+local json = require('json')
+local popen = require('popen')
+
+local t = require('luatest')
+local g = t.group()
+
+-- Find an item in an array table and return its index.
+--
+-- array_find('b', {'a', 'b', 'c'}) -> 2
+-- array_find('x', {'a', 'b', 'c'}) -> nil
+local function array_find(needle, array)
+    for i, item in ipairs(array) do
+        if item == needle then
+            return i
+        end
+    end
+    return nil
+end
+
+-- All Lua files produced by the test write a JSON object to
+-- stdout. This object contains everything we want to know
+-- about the context of the script execution.
+--
+-- First of all, we should ensure that given script was executed.
+-- So we output a script file name.
+--
+-- Next, we should define what is passed to the script as `...`
+-- (dots).
+--
+-- We should also inspect `arg` content to freeze certain
+-- behavior.
+local SCRIPT_TEMPLATE = [[
+print(require('json').encode({
+    ['script'] = '<script>',
+    ['...'] = {...},
+    ['arg[-1]'] = arg[-1],
+    ['arg[0]'] = arg[0],
+    ['arg[]'] = setmetatable(arg, {__serialize = 'seq'}),
```

Comment:
`arg` contains `arg[-1]` and `arg[0]`. `arg[]` is supposed to say 'arg as an array, from 1 to the end'.

I can change it to something else if `arg[]` makes things more confusing rather than self-descriptive. After my comment, do you think I should leave it or change (and to what)?

Reply:
We can leave at as is, but I suggest to add a comment about this. I missed a part, that `seq` serialization will omit `0` and `-1` indexes during JSON serialization. With comment it becomes clear.

Reply:
I have added a note:

```lua
-- Note: In order to serialize `arg` into JSON it is split into
-- `arg[-1]`, `arg[0]` and `arg[]`. The latter means an array of
-- all items from 1 to the end.
```